assets/js/components/ExecutionResults/ExecutionResults.test.jsx

```diff
@@ -1,60 +1,82 @@
 import React from 'react';
+import { screen } from '@testing-library/react';
+
 import { renderWithRouter } from '@lib/test-utils';
 import {
+  addCriticalExpectation,
+  addPassingExpectation,
   agentCheckResultFactory,
   checksExecutionCompletedFactory,
   checkResultFactory,
-  expectationResultFactory,
+  withEmptyExpectations,
 } from '@lib/test-utils/factories/executions';
 import { catalogFactory, hostnameFactory } from '@lib/test-utils/factories';
 
 import ExecutionResults from './ExecutionResults';
 
 const prepareStateData = (checkExecutionStatus) => {
-  const hostnames = hostnameFactory.buildList(2);
+  const hostnames = hostnameFactory.buildList(4);
   const [{ id: agentID1 }, { id: agentID2 }] = hostnames;
-  const agent1 = agentCheckResultFactory.build({ agent_id: agentID1 });
-  const agent2 = agentCheckResultFactory.build({ agent_id: agentID2 });
-  const agents = [agent1, agent2];
-  const checkResults = checkResultFactory.buildList(1, {
-    agents_check_results: [agent1, agent2],
+  const agentCheckResult1 = agentCheckResultFactory.build({
+    agent_id: agentID1,
   });
-  const executionResult = checksExecutionCompletedFactory.build({
-    check_results: checkResults,
+  const agentCheckResult2 = agentCheckResultFactory.build({
+    agent_id: agentID2,
+  });
+  const agentCheckResult3 = agentCheckResultFactory.build({
+    agent_id: agentID1,
+  });
+  const agentCheckResult4 = agentCheckResultFactory.build({
+    agent_id: agentID2,
+  });
+  let checkResult1 = checkResultFactory.build({
+    agents_check_results: [agentCheckResult1, agentCheckResult2],
     result: 'passing',
   });
 
+  checkResult1 = withEmptyExpectations(checkResult1);
+  checkResult1 = addPassingExpectation(checkResult1, 'expect');
+  checkResult1 = addPassingExpectation(checkResult1, 'expect_same');
+
+  let checkResult2 = checkResultFactory.build({
+    agents_check_results: [agentCheckResult3, agentCheckResult4],
+    result: 'critical',
+  });
+
+  checkResult2 = withEmptyExpectations(checkResult2);
+  checkResult2 = addPassingExpectation(checkResult2, 'expect');
+  checkResult2 = addCriticalExpectation(checkResult2, 'expect');
+
+  const executionResult = checksExecutionCompletedFactory.build({
```

Comment:
Here the order of things looks strange.
I think we should create `targets` before this, and assign it there, as the targets belongs to the execution result.

```
 const targets = [
    { agent_id: agentID1, checks: [checkID1, checkID2] },
    { agent_id: agentID2, checks: [checkID1, checkID2] },
  ];

const executionResult = checksExecutionCompletedFactory.build({
    check_results: [checkResult1, checkResult2],
    targets: targets,
    result: 'critical',
  });
```

After that, instead of creating a new execution data for the `lastExecution` element, we should directly pass this value.

```
  const lastExecution = {
    executionLoading: false,
    executionData: executionResult,
    error: '',
  };
```